src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,30 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        StringBuilder stringBuilder = new StringBuilder("Report for period "
+                + dateFrom + " - " + dateTo);
+        LocalDate firstDate = LocalDate.parse(dateFrom, formatter);
+        LocalDate secondDate = LocalDate.parse(dateTo, formatter);
+        for (String name: names) {
+            int salary = 0;
+            for (String dataLine: data) {
+                String[] elementOfData = dataLine.split(" ");
+                LocalDate date = LocalDate.parse(elementOfData[0], formatter);
```

Comment:
Consider checking if the 'date' is within the range including the 'dateFrom' and 'dateTo' after parsing it to avoid parsing it when it's not necessary. This could slightly improve performance by reducing the number of parsing operations.